### ISSUE TITLE
Some doubt about the optimize step of style encoder 

### ISSUE THREAD

**Issue by 980202006** [yl4579/StarGANv2-VC#14]
Hello, I have a little doubt about the optimization steps of style encoder. Whether self.optimizer.step('style_encoder', scaler=scaler) works, because input z_ref will not call style encoder.
![image](https://user-images.githubusercontent.com/24452502/138230241-fd08ba36-2864-42e3-9bd5-e78c0d2f3d14.png)

**Comment by yl4579** [comment]
These are the code copied directly from the original [stargan-v2] repo(https://github.com/clovaai/stargan-v2/blob/master/core/solver.py#L128), so it is not a mistake and it will learn the styles as intended. 

The idea here is that when training the mapping network, the style encoder is only made to learn the style that converts (reconstructs) the converted result back to the original domain, regardless of what domain it is converted to. When training the generator with the style encoder, the style encoder is fixed and it is used as a guide to diversifying the generated samples so that two different styles will produce different results. If you update the style encoder also in line 219, it will cause the style diversification loss to diverge because the style encoder can do anything to make the loss as large as possible.

**Comment by YoniLeibner** [comment]
Hii, I also noticed this case and I was wondering on the reason we need the ds loss (we want that s1 and s2 to result with different styles? so that the mel_ref will effect the result? if so did you tried to compute the loss also in line 219 just with lamda_ds=0?
